packages/server/src/oauth/utils.ts

```diff
@@ -980,3 +993,145 @@ async function tryAddOnBehalfOf(req: IncomingMessage | undefined, authState: Aut
   authState.onBehalfOf = onBehalfOf;
   authState.onBehalfOfMembership = onBehalfOfMembership;
 }
+
+/**
+ * Tries to authenticate the user using an external authentication provider.
+ * This function checks if the access token is a valid JWT and corresponds to an external authentication provider.
+ * If the token is valid, it retrieves the user's profile and project membership.
+ * If successful, it returns the auth state containing the login, project, membership, and user configuration.
+ * If the token is invalid or does not correspond to an external provider, it returns undefined.
+ *
+ * @param req - The incoming HTTP request.
+ * @param accessToken - The access token as provided by the client.
+ * @returns The auth state if the access token is valid and corresponds to an external authentication provider; otherwise, undefined.
+ */
+async function tryExternalAuth(req: Request | undefined, accessToken: string): Promise<AuthState | undefined> {
+  const externalAuthProviders = getConfig().externalAuthProviders;
+  if (!externalAuthProviders) {
+    // No external auth providers configured
+    return undefined;
+  }
+
+  if (!isJwt(accessToken)) {
+    // Not a JWT, so we cannot verify it
+    return undefined;
```

Comment:
Currently requiring JWT's, and not allowing opaque tokens.

In the future, we can revisit this decision.

In theory, we could have a "catch all" auth server for unrecognized access tokens.

I checked with the customer who is sponsoring this work, and they are ok with "JWT only" for now.